src/pages/workspaces/workspace/Notebooks.js

```diff
@@ -232,6 +250,14 @@ const Notebooks = _.flow(
     this.setState({ notebooks: _.reverse(_.sortBy('updated', notebooks)) })
   })
 
+  refreshApps = _.flow(
+    withErrorReporting('Error loading Apps'),
+    Utils.withBusyState(v => this.setState({ loading: v }))
```

Comment:
TOL: This adds another independent 'process' that writes to the `loading` state. It's worth considering what could happen if the two different refresh operations happened at the same time, or in close proximity (the end of one refresh might clear the spinner before the other refresh is done). In this case, it's unlikely, and the consequences seem fairly benign, so it's probably ok, but good to think about.

Reply:
Is worth thinking about for sure. I think probably relatively benign, so I am inclined to punt on it for now, if that is okay with you?